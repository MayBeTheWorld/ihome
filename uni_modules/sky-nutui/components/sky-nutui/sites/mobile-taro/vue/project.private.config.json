{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}