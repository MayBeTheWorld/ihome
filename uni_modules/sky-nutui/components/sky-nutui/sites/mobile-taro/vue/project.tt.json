{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
