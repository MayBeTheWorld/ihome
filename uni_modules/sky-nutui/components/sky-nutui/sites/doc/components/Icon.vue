<template>
  <div class="icon-moudle">
    <div class="item home" @click="goHome">
      <nut-icon name="home"></nut-icon>
    </div>
    <div class="item refresh" @click="refresh">
      <nut-icon name="refresh"></nut-icon>
    </div>
    <div class="item category" @click="toCategory($event)">
      <nut-icon name="category"></nut-icon>
    </div>
    <div class="qrcodepart" v-if="showCode" ref="codeRef" @click="stopClick($event)">
      <div class="qrcode"> </div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, reactive, ref, onMounted, toRefs } from 'vue';
import { useRoute, useRouter } from 'vue-router';
export default defineComponent({
  name: 'demo-icon',
  components: {},
  emits: ['refresh', 'goHome'],
  setup(props: any, { emit }: any) {
    const router = useRouter();
    const route = useRoute();

    const state = reactive({
      showCode: false
    });

    const codeRef = ref(null);

    onMounted(() => {
      document.addEventListener('click', (e: any) => {
        state.showCode = false; // 点击其他区域关闭
      });
      // console.log('route', route)
    });

    // 返回demo 首页页
    const goHome = () => {
      // router.push('/');
      emit('goHome');
    };

    // 刷新
    const refresh = () => {
      // router.go(0);
      emit('refresh');
    };

    // 二维码
    const toCategory = (e: any) => {
      state.showCode = !state.showCode;
      e.stopPropagation(); // 阻止事件向上冒泡
    };

    const stopClick = (e: any) => {
      e.stopPropagation(); // 阻止事件向上冒泡
    };

    return reactive({
      ...toRefs(state),
      codeRef,
      goHome,
      refresh,
      toCategory,
      stopClick
    });
  }
});
</script>
<style lang="scss" scoped>
.icon-moudle {
  display: flex;
  justify-content: flex-end;
  position: absolute;
  z-index: 10;
  left: 0;
  right: 0;
  bottom: 0;
  height: 40px;
  padding: 0 22px;
  color: #666;
  border-top: 1px solid #e6e6e6;
  background: #fff;
  .item {
    margin-left: 12px;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
  }
}
.qrcodepart {
  // display: none;
  position: absolute;
  right: 9px;
  bottom: 48px;
  width: 110px;
  padding: 6px 7px;
  border-radius: 6px;
  background: rgba(255, 255, 255, 1);
  box-shadow: 0px 4px 10px 0px rgba(0, 0, 0, 0.11);
  box-sizing: border-box;
  .qrcode {
    width: 96px;
    height: 96px;
    margin: 0 auto;
    background: url(https://img12.360buyimg.com/imagetools/jfs/t1/162421/39/13392/9425/6052ea60E592310a9/264bdff23ef5fe95.png)
      no-repeat;
    background-size: cover;
  }
}
</style>
