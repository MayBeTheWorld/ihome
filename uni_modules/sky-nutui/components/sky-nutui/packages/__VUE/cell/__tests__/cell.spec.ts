import { config, mount } from '@vue/test-utils';
import Cell from '../index.vue';
import Icon from '../../icon/index.vue';
import { nextTick } from 'vue';

beforeAll(() => {
  config.global.components = {
    [Icon.name]: Icon
  };
});

// mock module
jest.mock('vue-router', () => ({
  useRouter: jest.fn()
}));

test('prop title desc subtitle', async () => {
  const wrapper = mount(Cell, {
    props: {
      title: '标题1',
      desc: '描述1',
      subTitle: '副标题1'
    }
  });
  const html = expect(wrapper.html());
  html.toContain('标题1');
  html.toContain('描述1');
  html.toContain('副标题1');

  wrapper.setProps({
    title: '标题2',
    desc: '描述2',
    subTitle: '副标题2'
  });
  await nextTick();
  const html2 = expect(wrapper.html());
  html2.toContain('标题2');
  html2.toContain('描述2');
  html2.toContain('副标题2');

  html.toMatchSnapshot();
});

test('prop desc-text-align left', () => {
  const wrapper = mount(Cell, {
    props: {
      descTextAlign: 'left',
      desc: '张三'
    }
  });
  expect(wrapper.find<HTMLElement>('.nut-cell__value').element.style.textAlign).toEqual('left');
  expect(wrapper.html()).toMatchSnapshot();
});

test('prop isLink ', () => {
  const wrapper = mount(Cell, {
    props: {
      isLink: true
    }
  });
  expect(wrapper.find<HTMLElement>('.nut-cell__link'));
  expect(wrapper.html()).toMatchSnapshot();
});

test('emit click event', () => {
  const wrapper = mount(Cell);

  wrapper.trigger('click');
  expect(wrapper.emitted('click')!.length).toEqual(1);
});

test('slot default test', () => {
  const wrapper = mount(Cell, {
    props: {
      title: '标题1',
      desc: '描述1',
      subTitle: '副标题1'
    },
    slots: {
      default: 'Custom Content'
    }
  });
  expect(wrapper.html()).toContain('Custom Content');
  expect(wrapper.html()).toMatchSnapshot();
});

test('slot link、icon test', () => {
  const wrapper = mount(Cell, {
    props: {
      title: '标题1',
      desc: '描述1',
      subTitle: '副标题1'
    },
    slots: {
      icon: 'Custom Icon',
      link: 'Custom Link'
    }
  });
  const html = expect(wrapper.html());
  html.toContain('Custom Link');
  html.toContain('Custom Icon');
  expect(wrapper.html()).toMatchSnapshot();
});
