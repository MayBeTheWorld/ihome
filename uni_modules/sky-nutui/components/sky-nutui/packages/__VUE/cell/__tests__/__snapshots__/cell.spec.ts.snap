// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prop desc-text-align left 1`] = `
"<view class=\\"nut-cell\\" style=\\"border-radius: px;\\">
  <!--v-if-->
  <!--v-if-->
  <view class=\\"nut-cell__value nut-cell__value--alone\\" style=\\"text-align: left;\\">张三</view>
  <!--v-if-->
</view>"
`;

exports[`prop isLink  1`] = `
"<view class=\\"nut-cell nut-cell--clickable\\" style=\\"border-radius: px;\\">
  <!--v-if-->
  <!--v-if-->
  <!--v-if--><i class=\\"nutui-iconfont nut-icon nut-icon-right nut-cell__link\\" src=\\"\\"></i>
</view>"
`;

exports[`prop title desc subtitle 1`] = `
"<view class=\\"nut-cell\\" style=\\"border-radius: px;\\">
  <!--v-if-->
  <view class=\\"nut-cell__title\\">
    <view class=\\"title\\">标题1</view>
    <view class=\\"nut-cell__title-desc\\">副标题1</view>
  </view>
  <view class=\\"nut-cell__value\\" style=\\"text-align: right;\\">描述1</view>
  <!--v-if-->
</view>"
`;

exports[`slot default test 1`] = `"<view class=\\"nut-cell\\" style=\\"border-radius: px;\\">Custom Content</view>"`;

exports[`slot link、icon test 1`] = `
"<view class=\\"nut-cell\\" style=\\"border-radius: px;\\">
  <view class=\\"nut-cell__icon\\">Custom Icon</view>
  <view class=\\"nut-cell__title\\">
    <view class=\\"title\\">标题1</view>
    <view class=\\"nut-cell__title-desc\\">副标题1</view>
  </view>
  <view class=\\"nut-cell__value\\" style=\\"text-align: right;\\">描述1</view>Custom Link
</view>"
`;
